### PR TITLE
Write docs about our Docker build setup

### PR DESCRIPTION
**Pull request by pugnascotia**
Add a `README.md` to the docker distribution project, with details
about how we build and test our images.

### CODE REVIEW

File: distribution/docker/README.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+   3. Build the Docker image locally
+
+Some images use (1) as the releasable artifact, some use (3).
+
+**NOTE:** "Pipeline" isn't actually the correct term - in reality, each Gradle
+task depends on the one before it. Notably, it is the transform tasks that
+depend on a locally build `.tar.gz` Elasticsearch archive.
+
+
+## Releasing on Docker Hub
```
Comment (mark-vieira) [inline distribution/docker/README.md:46]: Should we include a link to https://github.com/elastic/dockerfiles somewhere in this section?

File: distribution/docker/README.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+Being an official image puts additional constraints on how the Elasticsearch
+image is built.
+
+   * It must extend another official image
+   * It must fetch any required artifacts - they cannot be supplied in the build
+     context.
+   * It must be platform-independent i.e. it can build on ARM and x64
+
+
+Step (2) in the build strategy above replaces the `curl` command in the
```
Comment (mark-vieira) [inline distribution/docker/README.md:67]: nit: We're referring to step "2" but the above list is a bulleted one. Should we make it an ordered list?

**Review by mark-vieira — APPROVED** [review]
Thanks Rory, this is super helpful. I added a couple minor comments but this otherwise LGTM.

**Review by breskeby — APPROVED** [review]

File: distribution/docker/README.md (hunk 2 above)
Comment (pugnascotia) [reply distribution/docker/README.md]: Actually it's referring to a list higher up the page. I've reworked it anyway and added a link.

File: distribution/docker/README.md (hunk 1 above)
Comment (pugnascotia) [reply distribution/docker/README.md:46]: Yes, good call.

**Pull request merged**